server/aesdsocket.c

```diff
@@ -248,123 +211,413 @@ int main(int argc, char *argv[])
+    syslog(LOG_INFO, "bind sucess \n");
 
-	     } while(buf[recv_bytes-1] != '\n');
             
-            /* STEP 8:
-           	 Returns the full content of /var/tmp/aesdsocketdata to the client as soon as the received data packet completes.
-            */
+    freeaddrinfo(servinfo);
+    servinfo = NULL;
+
+    return 0;
+}
+
+//function for timestamp thread to log timestamp
+void *timestamp_threadFunc(void *thread_param)
+{
+    thread_timestamp_t *threadDataTimeStamp = (thread_timestamp_t *)thread_param;
+    
+    time_t current_time ;
+    struct tm *local_time ;
+    char time_stamp_format[50];
+    struct timespec timeSpec;
+    int time_length;
+
+    syslog(LOG_INFO, "Timestamp thread started\n");
+
+    while(1)
+    {
+        // monotonic current time mode
+        if (clock_gettime(CLOCK_MONOTONIC, &timeSpec))
+        {
+            syslog(LOG_ERR, "clock_gettime failed \n");
+        }
+    
+        //add interval
+        timeSpec.tv_sec += threadDataTimeStamp->intervalSecs;
+
+        //sleep for interval
+        if (clock_nanosleep(CLOCK_MONOTONIC, TIMER_ABSTIME, &timeSpec, NULL))
+        {
+            syslog(LOG_ERR, "clock_nanosleep failed\n");
+        }
+
+        //get current time
+        time(&current_time);
+        local_time = localtime(&current_time);
+
+        //format current time
+        time_length = strftime(time_stamp_format, sizeof(time_stamp_format), "timestamp: %Y, %b %d, %H:%M:%S\n", local_time);
+
+        //lock mutex
+        if (pthread_mutex_lock(threadDataTimeStamp->thread_mutex) == -1)
+        {
+            syslog(LOG_ERR, "mutex lock for timestamp failed \n");
+        }
+
+        //write data to file
+        status = write(logfile_fd, time_stamp_format, time_length);
+        if (status == -1)
+        {
+            syslog(LOG_ERR, "timestamp write to file failed: %s\n", strerror(errno));
+        }
+
+
+        //unlock mutex
+        if (pthread_mutex_unlock(threadDataTimeStamp->thread_mutex) == -1)
+        {
+            syslog(LOG_ERR, "mutex unlock for timestamp failed \n");
+        }
+
+    }
+}
+
+//function to setup timestamp
+void timestamp_setup (void)
+{
+    thread_timestamp.thread_mutex = &mutex;
+    thread_timestamp.intervalSecs =10;
+
+    status = pthread_create(&(thread_timestamp.thread_id), NULL, \
+                                timestamp_threadFunc, &(thread_timestamp));
+    if(status == -1)
+    {
+        syslog(LOG_ERR, "Timestamp Thread creation failed");
+    }
+
+}
+
+
+void *recv_send_thread(void *thread_param)
+{
+    memset(buf, 0, BUFFER_LEN);
+    memset(read_buf, 0, BUFFER_LEN);
+
+    thread_data_t *thread_data = (thread_data_t*)thread_param;
+    
+    /*STEP 6:
+        Logs message to the syslog “Accepted connection from xxx” 
+    */
+    inet_ntop(thread_data->client_addr->ss_family, get_in_addr((struct sockaddr *)&(thread_data->client_addr)), s, sizeof s);  // s is the ip address
+    syslog(LOG_INFO,"Accepted connection from %s",s);
```

Comment:
@ishassharmaa since `s` is global what would happen if two different accept threads were started from two different connections at the same time?